NSSAClient.java

```diff
@@ -8,20 +8,59 @@
  */
 
 public class NSSAClient {
-
+	
+	Scanner in = null;
     /**
      * Main
      */
     public static void main(String[] args){
+    	InetAddress a;
+		try {
+			InetAddress hostName = InetAddress.getLocalHost();
+			String a1 = hostName.getHostAddress();
+			
+			System.out.println(a1);
+			System.out.println(hostName);
+			System.exit(0);
+		} catch (UnknownHostException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		
         new NSSAClient();
+        
     }
 
     /**
      * Constructor
      */
     public NSSAClient(){
+    	in = new Scanner(System.in);
         // Create process for command line interface here
-
+    	System.out.println("Enter UDP or TCP:");
+    	String type = in.nextLine();
+    	while(!(type.toLowerCase().equals("tcp") || type.toLowerCase().equals("udp"))) {
+    		System.out.println("Enter UDP or TCP:");
+    		type = in.nextLine();
+    	}
+    	
+    	System.out.println("Enter port to connect:");
+    	int port = in.nextInt();
+    	
+//    	change Server's validPort() to public, so I don't have to rewrite...lol
+    	while(!(validatePort(port))) {
+    		System.out.println("Enter port to connect:");
+    		port = in.nextInt();
+    	}
+    	
+    	switch(type) {
+	    	case "udp":
+	    			Udp(port);
```

Comment:
So you would need to make two more files: TCPClient and UDPClient. This file, NSSAClient, will act as the main file that will first talk with the user. So when you split udp and tcp into their own files, you'd call something like: 
TCPClient tcpc = new TCPClient(port)  AND UDPClient udpc = new UDPClient(port) and these class files will handle the communication.